fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java

```diff
@@ -0,0 +1,239 @@
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.stream.Collectors.toSet;
+
+public class IcebergTableStatisticCalculator {
+    private static final Logger LOG = LogManager.getLogger(IcebergTableStatisticCalculator.class);
+
+    private final Table icebergTable;
+
+    private IcebergTableStatisticCalculator(Table icebergTable) {
+        this.icebergTable = icebergTable;
+    }
+
+    public static Statistics getTableStatistics(List<Expression> icebergPredicates,
+                                                Table icebergTable,
+                                                Map<ColumnRefOperator, Column> colRefToColumnMetaMap) {
+        return new IcebergTableStatisticCalculator(icebergTable)
+                .makeTableStatistics(icebergPredicates, colRefToColumnMetaMap);
+    }
+
+    private Statistics makeTableStatistics(List<Expression> icebergPredicates,
+                                           Map<ColumnRefOperator, Column> colRefToColumnMetaMap) {
+        LOG.debug("Begin to make iceberg table statistics!");
+        Optional<Snapshot> snapshot = IcebergUtil.getCurrentTableSnapshot(icebergTable, true);
+        if (!snapshot.isPresent()) {
+            return Statistics.builder().build();
+        }
+
+        List<Types.NestedField> columns = icebergTable.schema().columns();
+
+        Map<Integer, Type.PrimitiveType> idToTypeMapping = columns.stream()
+                .filter(column -> column.type().isPrimitiveType())
+                .collect(Collectors.toMap(Types.NestedField::fieldId, column -> column.type().asPrimitiveType()));
+        List<PartitionField> partitionFields = icebergTable.spec().fields();
+
+        Set<Integer> identityPartitionIds = IcebergUtil.getIdentityPartitions(icebergTable.spec()).keySet().stream()
+                .map(PartitionField::sourceId)
+                .collect(toSet());
+
+        List<Types.NestedField> nonPartitionPrimitiveColumns = columns.stream()
+                .filter(column -> !identityPartitionIds.contains(column.fieldId()) && column.type().isPrimitiveType())
+                .collect(toImmutableList());
+
+        Map<Integer, String> idToColumnNames = columns.stream()
+                .filter(column -> column.type().isPrimitiveType())
+                .collect(Collectors.toMap(Types.NestedField::fieldId, column -> column.name()));
+
+        TableScan tableScan = IcebergUtil.getTableScan(icebergTable,
+                snapshot.get(), icebergPredicates, true);
+
+        IcebergFileStats icebergFileStats = null;
+        try (CloseableIterable<FileScanTask> fileScanTasks = tableScan.planFiles()) {
```

Comment:
does it mean, this is an extra plan here, only for the purpose of stats calculation.